snuba/query/schema.py

```diff
@@ -145,31 +146,54 @@
         # Conditions at the second level are ORed together.
         # eg: [(a, =, 1), (b, =, 2)] => "a = 1 AND b = 2"
         # eg: [(a, =, 1), [(b, =, 2), (c, =, 3)]] => "a = 1 AND (b = 2 OR c = 3)"
-        "condition": {
+        "condition_lhs": {
+            "anyOf": [
+                {"$ref": "#/definitions/column_name"},
+                {"$ref": "#/definitions/nested_expr"},
+            ],
+        },
+        "binary_condition": {
             "type": "array",
             "items": [
+                {"$ref": "#/definitions/condition_lhs"},
                 {
+                    # Operator
+                    "type": "string",
+                    "enum": BINARY_OPERATORS,
+                },
+                {
+                    # Literal
```

Comment:
Is there a reason we lock the RHS to a literal instead of another column or expression?

Reply:
Yes, in the legacy query language, a condition is `<column>|<function> <operator> <literal>`   . Only literals are allowed on the rhs. The reason why the language is designed this way is unknown to me, but I bet it was to make processing simpler.